src/cclib/io/cmlwriter.py

```diff
@@ -43,6 +43,9 @@ def generate_repr(self):
         if self.jobfilename is not None:
             d['id'] = self.jobfilename
         _set_attrs(molecule, d)
+        
+        if hasattr(self.ccdata, 'atomnos'):
+            elements = [self.pt.element[Z] for Z in self.ccdata.atomnos]
```

Comment:
Is ```elements``` only used when writing the ```atomArray```? If so, move it just above the ```for atomid ...``` and combine the two ```if hasattr...``` statements.